### PR TITLE
Change from click to argparser

### PR DESCRIPTION
**Pull request by migueLib**
Hi Yosuke, I removed the name function and replace click with argparse, I also added some documentation to the code.

### CODE REVIEW

**Comment by yfukai** [comment]
Hi, thanks @migueLib for this PR, just wondering, but could you give me the reasoning for the `click -> argparse` migration? Is it to reduce the dependency, or is it easier to maintain with the `argparse`?

**Comment by yfukai** [comment]
If you're willing to update the script, I'm happy to add you as a collaborator of the repository.

File: main.py [hunk 1]
```diff
@@ -157,15 +136,22 @@ def main(
...
-    output_folder = Path(output_folder)
-    input_path2 = get_main_name(input_path)
-    flatfield_path = output_folder / (input_path2 + "-ffp.tiff")
-    darkfield_path = output_folder / (input_path2 + "-dfp.tiff")
+    # Re-arrange flatfields and darkfields axis to be (z, y, x, c)
     flatfields = np.moveaxis(np.array(flatfields), 0, -1)
     darkfields = np.moveaxis(np.array(darkfields), 0, -1)
+
+    # Get output file names, splitext gets the file name without the extension
+    flatfield_path = args.output_folder / f"{splitext(args.input.name)[0]}-ffp.tiff"
```
Comment (yfukai) [inline main.py:144]: This gives a slightly different output from the original version, but I believe it's OK.

**Comment by yfukai** [comment]
Let me check it with test data and I'll be back to you!

**Comment by migueLib** [comment]
Dear @yfukai. Yes! For a more stable tool I'd say is best practice to reduce the number of dependencies as much as posible 😃. Both click and argpsrse are great for creating a CLI. But the fact that is a third party library makes it an extra dependency to deal with.

**Comment by yfukai** [comment]
I wrongly merged before fixing problems, I'll fix by myself 😓

**Comment by yfukai** [comment]
Now fixed!

**Pull request merged**